op-node/l2/source.go

```diff
@@ -88,24 +89,12 @@ func (s *Source) ForkchoiceUpdate(ctx context.Context, fc *eth.ForkchoiceState,
 		} else {
 			e.Error("Failed to share forkchoice-updated signal")
 		}
-	}
-	switch result.PayloadStatus.Status {
-	case eth.ExecutionSyncing:
-		return nil, fmt.Errorf("updated forkchoice, but node is syncing: %v", err)
-	case eth.ExecutionAccepted, eth.ExecutionInvalidTerminalBlock, eth.ExecutionInvalidBlockHash:
-		// ACCEPTED, INVALID_TERMINAL_BLOCK, INVALID_BLOCK_HASH are only for execution
-		return nil, fmt.Errorf("unexpected %s status, could not update forkchoice: %v", result.PayloadStatus.Status, err)
-	case eth.ExecutionInvalid:
-		return nil, fmt.Errorf("cannot update forkchoice, block is invalid: %v", err)
-	case eth.ExecutionValid:
-		return &result, nil
-	default:
-		return nil, fmt.Errorf("unknown forkchoice status on %s: %q, ", fc.SafeBlockHash, string(result.PayloadStatus.Status))
+		return nil, err
```

Comment:
This does change the error behavior. You should call `eth.ForkchoiceUpdateErr` to transform the status into an error to preserve the current behavior. In the `batch-derivation` branch it changes again though, the difference between an invalid payload and a RPC connection error shouldn't be hidden from the derivation code.

Reply:
done